### DISCUSSION TITLE
Error being thrown into the DOM when trying to create components via JavaScript

### DISCUSSION THREAD

**Discussion by jordanjones243** [AlaskaAirlines/WC-Generator#529] [General Auro discussions]
An issue was brought to my attention recently with creating an instance of `auro-popover` through JavaScript that likely can come up in other components:
- https://github.com/AlaskaAirlines/auro-popover/issues/69

Basically, in most of our components, we have attributes that are applied to our custom elements. These attributes are usually initialized inside of our constructor and some are reflected onto the element in the DOM. Where the issue arises is when we try to create a component using the DOM (via JavaScript). An error is thrown because the DOM is complaining that it cannot create an element that already has attributes applied to it. In this case, we had our `auro-popover` component being created with an `id` attribute already being assigned a value in the constructor. 

What we need to do is go through all of our components and move all of our attributes that are being reflected onto the components from being initialized in the constructor to the `connectedCallback()` function, which runs once an element is attached to the DOM.

This simple change from `auro-popover` can be used as a guide to make the change in any other component that can run into this issue:
- https://github.com/AlaskaAirlines/auro-popover/pull/68/files

|repo|supports dynamic DOM|comments
|---|---|---
|auro-accordion|
|auro-alert|
|auro-avatar|
|auro-background|
|auro-backtotop|
|auro-badge|
|auro-banner|
|auro-button|
|auro-card|
|auro-carousel|
|auro-checkbox|
|auro-combobox|
|auro-datetime|
|auro-datepicker|
|auro-dialog|
|auro-drawer|
|auro-dropdown|
|auro-formvaliator|
|auro-flight|
|auro-flightline|
|auro-header|
|auro-hyperlink|
|auro-icon|
|auro-input|
|auro-loader|
|auro-lockup|
|auro-menu|
|auro-nav|
|auro-pane|
|auro-popover|
|auro-radio|
|auro-select|
|auro-skeleton|
|auro-sidenav|
|auro-table|
|auro-toast|

**Comment by blackfalcon** [comment]
New issue with similar performance feature request 

- https://github.com/AlaskaAirlines/auro-input/issues/300

**Comment by Patrick-Daly-AA** [comment]
TODO Items to address this:
1) Make a list of all the components which need this update
2) Add test to baseline test quite in generator to catch this issue in the future.

**Reply by blackfalcon** [reply]
@jordanjones243 are these items something you are working on?

**Reply by jordanjones243** [reply]
@blackfalcon I have been adding this test to components as I have worked on. Mainly, the form components have gotten them added to their draft PRs with the form validation work I have been doing.

**Reply by blackfalcon** [reply]
@jordanjones243 can you please list here what components you have updated and what ones need to be updated? I have added a table to the discussion for tracking.
